tiled/adapters/zarr.py

```diff
@@ -49,16 +50,28 @@ def init_storage(
 
         """
         data_source = copy.deepcopy(data_source)  # Do not mutate caller input.
-        data_uri = storage.uri + "".join(
-            f"/{quote_plus(segment)}" for segment in path_parts
-        )
         # Zarr requires evenly-sized chunks within each dimension.
         # Use the first chunk along each dimension.
         zarr_chunks = tuple(dim[0] for dim in data_source.structure.chunks)
         shape = tuple(dim[0] * len(dim) for dim in data_source.structure.chunks)
-        directory = path_from_uri(data_uri)
-        directory.mkdir(parents=True, exist_ok=True)
-        store = zarr.storage.DirectoryStore(str(directory))
+        if storage.bucket:
```

Comment:
In the old model, the adapter was handed multiple storage options and had to pick the one it wanted. Now, the _caller_ in `tiled.catalog.adapter` picks _one_ storage option and passes just that one in.

https://github.com/bluesky/tiled/blob/f3331ef90de056e671ba5acb0a80350c473a696d/tiled/catalog/adapter.py#L672-L695

So, the task here is to check `isinstance(storage, BucketStorage)` versus `FileStorage`. Additionally, the `supported_storage` attribute on this class should be extended to include `BucketStorage`. This tells the caller to offer `BucketStorage` if that is the highest-priority item in `writable_storage`.